renku/api/datasets.py

```diff
@@ -51,6 +51,26 @@ def renku_datasets_path(self):
         """Return a ``Path`` instance of Renku dataset metadata folder."""
         return self.renku_path.joinpath(self.DATASETS)
 
+    def datasets_from_commit(self, commit=None):
```

Comment:
Can we add a comment about which type is `commit`?

Reply:
Do you want to start including type references everywhere?

Reply:
It helps me while reading the code and since I'm not super familiar with those types within git library, it would save me some time.

Not necessary everywhere.

Reply:
We can do a separate documentation sprint.

In this case, I am not a big fan of adding parameters docs just for a sake of having them if they are self explanatory:

```python
    """Return datasets defined in a commit.

    :param commit: ???
    :type commit: git.Commit
    """
```

I would rather see:

```python
    def datasets_from_commit(self, commit: Optional[git.Commit] = None) -> Generator[Dataset]:
```

Reply:
:+1: